libs/datamodel/core/src/ast/reformat/reformatter.rs

```diff
@@ -362,9 +383,65 @@ impl<'a> Reformatter<'a> {
         }
     }
 
+    fn extract_and_sort_directives<'i>(token: &'i Token, is_field_directive: bool) -> Vec<Pair<'i, Rule>> {
+        // get indices of directives and store in separate Vector
+        let mut directive_indices = Vec::new();
+        let mut directives = Vec::new();
+        for (i, pair) in token.clone().into_inner().enumerate() {
+            match pair.as_rule() {
+                Rule::directive => {
+                    directive_indices.push(i);
+                    directives.push(pair);
+                }
+                _ => (),
+            }
+        }
+
+        // sort directives
```

Comment:
Can you add to this comment that it must match the order in the other file please?